lib/thrift/binary/framed/client.ex

```diff
@@ -286,15 +311,51 @@ defmodule Thrift.Binary.Framed.Client do
       :ok ->
         {:noreply, s}
 
+      {:error, :closed} = error ->
+        if reconnect do
+          {:disconnect, {:reconnect, {:cast, msg}}, s}
+        else
+          {:disconnect, error, s}
+        end
+
       {:error, _} = error ->
         {:disconnect, error, s}
     end
   end
 
+  @impl Connection
+  def handle_info({:tcp_closed, sock}, %{reconnect: true, sock: {_transport, sock}} = s) do
```

Comment:
Handle `:ssl_closed` too? I think we can disconnect on tcp_error or ssl_error too. In fact if we get data we probably want to error to crash.

I think we want to disconnect on this always, as we are outside a request and so a disconnect allows us to reconnect.